src/frontend/src/utils/overwrite_options.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::handler::HandlerArgs;
+
+#[derive(Debug, Clone, Default)]
+pub struct OverwriteOptions {
```

Comment:
This is for overwriting MView's properties only, right? How about renaming to "MviewOptions" and putting to somewhere closer to MV instead of `frontend::utils`?

Reply:
Not sure if we should also apply it to the sink. I prefer the current implementation which stores the immutable with options in the common context and all stream-optimizing parts share a common and consistent value. :thinking:

Reply:
> This is for overwriting MView's properties only, right? How about renaming to "MviewOptions" and putting to somewhere closer to MV instead of frontend::utils?

+1 for `MviewOptions`, `OverwriteOptions` sounds a little vague.

Reply:
> Not sure if we should also apply it to the sink. I prefer the current implementation which stores the immutable with options in the common context and all stream-optimizing parts share a common and consistent value. 🤔

The current approach only sets `STREAMING_RATE_LIMIT` per session, so actually it may not be "common and consistent".

This PR builds on the current approach, by actually associating the option with an MV, and allowing to overwrite it in the MV with `with_options`.

I think this approach is needed, to throttle specific problematic MVs, rather than all MVs.

Reply:
> The current approach only sets STREAMING_RATE_LIMIT per session, so actually it may not be "common and consistent".

But it can also be setted when creating any streaming jobs such as creating a sink. If a code branch or ddl handler need some options, they should handle them and add it in the context.

Reply:
Bump, let's update it to `MviewWithOptions`. Eventually we should also refactor all `with` options to be specific to the DDL statements.

Sink will also have rate limit, but it should be contained in `SinkWithOptions`.

Tracking issue: https://github.com/risingwavelabs/risingwave/issues/13061

Reply:
> > The current approach only sets STREAMING_RATE_LIMIT per session, so actually it may not be "common and consistent".
> 
> But it can also be setted when creating any streaming jobs such as creating a sink. If a code branch or ddl handler need some options, they should handle them and add it in the context.

If we have some concept for streaming jobs (including mviews & sinks), it would be better.

Otherwise, I think `MviewWithOptions` is still better than `OverwriteOptions`, which essentials means nothing to me.

+1 for #13061

Reply:
> Bump, let's update it to MviewWithOptions. Eventually we should also refactor all with options to be specific to the DDL statements.
> 
> Sink will also have rate limit, but it should be contained in SinkWithOptions.

I'd like to keep the current naming and leave it to future decisions because `OverwriteOptions` is a member of `OptimizerContext`, which applies to all `handle_create_*` functions, including mv, source, and sink. So I doubt whether there will be a `SinkWithOptions` or not.